nsxt/resource_nsxt_policy_tier0_gateway_gre_tunnel.go

```diff
@@ -0,0 +1,408 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s/locale_services"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyTier0GatewayGRETunnel() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyTier0GatewayGRETunnelCreate,
+		Read:   resourceNsxtPolicyTier0GatewayGRETunnelRead,
+		Update: resourceNsxtPolicyTier0GatewayGRETunnelUpdate,
+		Delete: resourceNsxtPolicyTier0GatewayGRETunnelDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyTier0GatewayGRETunnelImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"locale_service_path": {
+				Type:        schema.TypeString,
+				Description: "Policy path of associated Gateway Locale Service on NSX",
+				Required:    true,
+			},
+			"destination_address": {
+				Type:         schema.TypeString,
+				Description:  "Destination IPv4 address",
+				Required:     true,
+				ValidateFunc: validateSingleIP(),
```

Comment:
If this is IPv4 only, there is `validation.IsIPv4Address` helper. Not sure the description is accurate though